cpp/cmake/thirdparty/get_hnswlib.cmake

```
@@ -64,6 +64,10 @@ function(find_and_configure_hnswlib)
         GLOBAL_TARGETS hnswlib
         NAMESPACE hnswlib::)
     endif()
+    include("${rapids-cmake-dir}/export/find_package_root.cmake")
+    rapids_export_find_package_root(
+      BUILD hnswlib [=[${CMAKE_CURRENT_LIST_DIR}]=] EXPORT_SET raft-exports
+    )
```

Alternatively, we could include the build `rapids_export` outside of this conditional and then export the root unconditionally, perhaps that's more correct. WDYT?
```suggestion
      include("${rapids-cmake-dir}/export/find_package_root.cmake")
      rapids_export_find_package_root(
        BUILD hnswlib [=[${CMAKE_CURRENT_LIST_DIR}]=] EXPORT_SET raft-exports
      )
    endif()
```

---

@vyasr yes, I think you're right. The `rapids_export(BUILD hnswlib)` should be moved outside the `if(NOT PKG_EXCLUDE_FROM_ALL)` block.